Add support for buildPythonPackage (#15)
jtojnar/nixpkgs-hammering: explanations/python-inconsistent-interpreters.md

Each python derivation in nixpkgs should take its inputs in such a way that it
can be properly built for multiple different python interpreters.

For example, the following is incorrect:

```
{ lib
, buildPythonPackage
, python3Packages
}:

buildPythonPackage rec {
  pname = "mypackage";
  ...
  propagatedBuildInputs = [ python3Packages.numpy ];
}
```

Instead, it should be:

```
{ lib
, buildPythonPackage
, numpy
}:

buildPythonPackage rec {
  pname = "mypackage";
  ...
  propagatedBuildInputs = [ numpy ];
}
```

> This design will allow nixpkgs to ensure that the version of python pulled in by `numpy` is the same as the version of python that the package is built for.

Reviewer: ```suggestion
This design will allows nixpkgs to ensure that the version of python pulled in by `numpy` is the same version that the package uses.
```
